invisible_cities/database/load_db_test.py

```diff
@@ -178,3 +179,26 @@ def test_pmt_noise_frequencies(db):
     assert frequencies.shape[0] == db.nfreqs
     assert frequencies.shape[1] == db.feboxes + 1
     assert np.all(frequencies[:, 0] == freq_expected)
+
+
+def test_RadioactivityData_max_version():
+    '''Check that RadioactivityData returns by default the data with maximum Version number'''
+
+    db_file = "next100"
+
+    # test unique (G4Volume, Isotope)
+    df, _ = DB.RadioactivityData(db_file)
+    assert np.all(df.groupby(["G4Volume", "Isotope"]).count() == 1)
+
+    # test TotalActivity values correspond to maximum version number
+    conn = sqlite3.connect(DB.get_db(db_file))
+
+    query = '''SELECT * FROM Activity
+               WHERE G4Volume = '{0}' AND Isotope = '{1}'
+            '''
+
+    for (g4volume, isotope), full in df.groupby(["G4Volume", "Isotope"]):
+        req = pd.read_sql_query(query.format(g4volume, isotope), conn)
+        req = req[req.Version == req.Version.max()]
+
+        np.testing.assert_allclose(req.TotalActivity, full.TotalActivity)
```

Comment:
The naming is the other way around, right? `df` contains the data you requested (the one you are testing). `pd.read_sql_query(query.format(g4volume, isotope), conn)` gives you the full output, not filtered by version. Perhaps `full` wasn't the best name suggestion, how about
```suggestion
    for (g4volume, isotope), req in df.groupby(["G4Volume", "Isotope"]):
        all_versions = pd.read_sql_query(query.format(g4volume, isotope), conn)
        max_version = all_versions[all_versions.Version == all_versions.Version.max()]

        np.testing.assert_allclose(req.TotalActivity, max_version.TotalActivity)
```

Also, why the groupby? Is it because the data may come in different orders? If so, you may want to sort both dataframes and avoid the loop using `assert_dataframes_equal`.

Finally, this test checks that without version number you get the highest one, and that's great, but you should write another one forcing a specific version number and check that you get the correct result.

Reply:
Not sure what you mean with different data orders. The objective is to avoid loading the whole database table, first selecting the data for a single G4Volume and Isotope and then explicitly keep the TotalActivity of the highest version number. This tests whether the sql-query for the whole database table in the `RadioactivityData` function is doing what we want.

I assumed that the test for a version number is uneccessary since the query in `RadioactivityData` is the same and we may only have a `version=0` (which may be unchanged for some time) once a non-dummy database is loaded. I can add the test for `version=0` if you found it neccessary now, or wait for changed in the database in the future.

Reply:
> The objective is to avoid loading the whole database table, 

Ok. Loading the full table is probably a non-issue, but I agree that there is no need to do that. I was thinking that requesting so many queries to mysql could make it run slow. If that's not the case I'm ok with it.

>  I can add the test for version=0 if you found it neccessary now, or wait for changed in the database in the future.

It doesn't take long and we will probably forget about adding that test, so better do it now that we are focused on it.

We need to make it work for many versions so we I would add also version 2 updating other values than in version 1 to the activity table and eventually do the same with the efficiency table.

Once we do this with a dummy database, we can sleep in peace knowing that the code is ready to handle the real database.

Reply:
I have added the test for Efficiency table and for the version number. The new test does it for `version = 0`, otherwise, for `version > 0` may give future problem since the first non-dummy database version could be left unmodified during some time at `version = 0` until some update is needed.

I am not aware if the sql querys are slower (my unfounded opinion would say that no, since sqlite should be designed to allow fast data selection), but the tests are fast anyway.

Reply:
My point was actually that we should add a test (or many) requesting versions 0, 1, and 2 _now_ because we have this dummy database with multiple versions. Once we load the real database we won't have multiple versions so the tests won't tell us much. Nonetheless, we will be confident that it works because we tested it before with this dummy database.

In principle, the test shouldn't fail for versions 1 and 2 even if they don't exist in the database because it should take the last available version. Therefore, I think you can simply parametrize it.